tests/manage/mcg/test_noobaa_secret.py

```diff
@@ -0,0 +1,332 @@
+
+    def finalizer():
+        for instance in instances[::-1]:
+            try:
+                instance.delete()
+            except CommandFailed as ex:
+                if "not found" in str(ex).lower():
+                    logger.warning(
+                        f"Resource {instance.name} could not be found in cleanup."
+                        "\nSkipping deletion."
+                    )
+                else:
+                    raise
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+class TestNoobaaSecrets:
+    def test_duplicate_noobaa_secrets(
+        self, backingstore_factory, cloud_uls_factory, mcg_obj, teardown_factory
+    ):
+        """
+        Objective of this test is:
+            1) Create a secret with the same credentials and see if the duplicates are allowed when BS created
+            2) Modify the existing secret credentials see if the owned BS/NS is getting reconciled
+
+        """
+        # create secret with the same credentials to check if duplicates are allowed
+        first_bs_obj = backingstore_factory(
+            method="oc", uls_dict={"aws": [(1, "eu-central-1")]}
+        )[0]
+        try:
+            logger.info(
+                "Trying to load credentials from ocs-ci-data. "
+                "This flow is only relevant when running under OCS-QE environments."
+            )
+            secret_dict = update_config_from_s3().get("AUTH")
+        except (AttributeError, EndpointConnectionError):
+            logger.warning(
+                "Failed to load credentials from ocs-ci-data.\n"
+                "Your local AWS credentials might be misconfigured.\n"
+                "Trying to load credentials from local auth.yaml instead"
+            )
+            secret_dict = load_auth_config().get("AUTH", {})
+        aws_client = S3Client(auth_dict=secret_dict["AWS"])
+        teardown_factory(aws_client.secret)
+        logger.info(f"New secret created: {aws_client.secret.name}")
+
+        cloud = "aws"
+        uls_tup = (1, "eu-central-1")
+        uls_name = list(cloud_uls_factory({cloud: [uls_tup]})["aws"])[0]
+        logger.info(f"ULS dict: {type(uls_name)}")
+        second_bs_name = create_unique_resource_name(
+            resource_description="backingstore",
+            resource_type=cloud.lower(),
+        )
+        bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
+        bs_data["metadata"]["name"] = second_bs_name
+        bs_data["metadata"]["namespace"] = config.ENV_DATA["cluster_namespace"]
+        bs_data["spec"] = {
+            "type": "aws-s3",
+            "awsS3": {
+                "targetBucket": uls_name,
+                "region": "eu-central-1",
+                "secret": {
+                    "name": aws_client.secret.name,
+                    "namespace": bs_data["metadata"]["namespace"],
+                },
+            },
+        }
+        second_bs_obj = create_resource(**bs_data)
+        teardown_factory(second_bs_obj)
```

Comment:
seems strange to call the test teardown as a test step

Reply:
@ebenahar this is special scenario where we can't use fixtures. so teardown_factory() fixture would help us add the objects to the teardown queue and will be taken care at the teardown phase of the test not during the test phase.